modin/pandas/dataframe.py

```diff
@@ -1692,9 +1720,22 @@ def query(self, expr, inplace=False, **kwargs):  # noqa: PR01, RT01, D200
         """
         Query the columns of a ``DataFrame`` with a boolean expression.
         """
-        self._update_var_dicts_in_kwargs(expr, kwargs)
+        # Remove `level` from the kwargs if it's already there. "level" doesn't
+        # make sense within the remote execution context, where the remote
+        # functions have a stack which is different from the stack on the
+        # driver node that ends in the modin eval call.
+        level = kwargs.pop("level", 0)
+
+        # Bump level up one because this function is wrapped in the Modin
+        # logging function wrapper. The alternative is for the wrapper to
+        # give the Modin functions that deal with `level` the special
+        # treatment of bumping `level` up by one, but that's not so nice
+        # eitiher.
+        level += 1
```

Comment:
given I see exact same wall of comments three times already, maybe we should turn this into a helper function like

```python
def _extract_eval_level(kwargs):
    """
    Extract `level` from eval or query `kwargs`.

    Remove `level` from the kwargs if it's already there. "level" doesn't
    make sense within the remote execution context, where the remote
    functions have a stack which is different from the stack on the
    driver node that ends in the modin eval call.

    Bump level up one because eval/query are wrapped in the Modin
    logging function wrapper.
    """
    level = kwargs.pop("level", 0) + 1
    return level, kwargs
```
and then use it here like

```python
def query(self, expr, inplace=False, **kwargs):
    # docstring
    level, kwargs = _extract_eval_level(kwargs)
    # do the rest of the stuff
```